### ISSUE TITLE
[Feature Request]: Support relational Joins in Python SDK

### ISSUE THREAD

**Issue by jcalvarezj** [apache/beam#24148]
### What would you like to happen?

The only current way to directly implement relational joins between two PCollections using the library is through Java SDK's org.apache.beam.sdk.schemas.transforms.Join. As a Python pipeline developer I would like to be able to do the same, in a similar fashion as described in the usage examples of https://beam.apache.org/documentation/sdks/java-extensions/#join-library

.add-labels python,'new feature'

### Issue Priority

Priority: 2

### Issue Component

Component: sdk-py-core

**Comment by kennknowles** [comment]
I believe the main way you can do this in Python is directly using [`CoGroupByKey`](https://beam.apache.org/releases/pydoc/2.42.0/apache_beam.transforms.util.html?highlight=cogroupbykey#apache_beam.transforms.util.CoGroupByKey)

Your request is to have a friendly wrapper for an equijoin on a particular schema field of two schema-aware PCollections? This seems like a possible project as a first contribution. A bit advanced, but doable.

CC @tvalentyn

**Comment by jcalvarezj** [comment]
> I believe the main way you can do this in Python is directly using [`CoGroupByKey`](https://beam.apache.org/releases/pydoc/2.42.0/apache_beam.transforms.util.html?highlight=cogroupbykey#apache_beam.transforms.util.CoGroupByKey)
> 
> Your request is to have a friendly wrapper for an equijoin on a particular schema field of two schema-aware PCollections? This seems like a possible project as a first contribution. A bit advanced, but doable.
> 
> CC @tvalentyn

That's right, I've worked with CoGroupByKey successfully with a couple workarounds, but I feel that schema-aware PCollections Join operations would save a lot of repetitive work on many use cases (as it's a more direct and friendly way as far as I've seen in Java implementations)
